frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/ResetCurrentToRemoteBackgroundable.java

```diff
@@ -29,23 +28,23 @@
 @ExtensionMethod({GitMacheteBundle.class})
 // For some reason, using `@ExtensionMethod({GitVfsUtils.class})` on this class
 // leads to weird Checker Framework errors :/
-public class ResetCurrentToRemoteBackgroundable extends Task.Backgroundable {
+public class ResetCurrentToRemoteBackgroundable extends SideEffectingBackgroundable {
 
   private final String localBranchName;
   private final String remoteTrackingBranchName;
   private final GitRepository gitRepository;
 
   public ResetCurrentToRemoteBackgroundable(String title,
       String localBranchName, String remoteTrackingBranchName, GitRepository gitRepository) {
-    super(gitRepository.getProject(), title);
+    super(gitRepository.getProject(), title, "reset");
     this.localBranchName = localBranchName;
     this.remoteTrackingBranchName = remoteTrackingBranchName;
     this.gitRepository = gitRepository;
   }
 
   @Override
   @UIThreadUnsafe
-  public void run(ProgressIndicator indicator) {
+  public void doRun(ProgressIndicator indicator) {
     if (myProject != null && localBranchName != null && remoteTrackingBranchName != null) {
```

Comment:
maybe let's refer to the `SideEffectingBackgroundable`'s `project` field instead? and let's be consistent about it? 🤔

Reply:
Right, good spot...